client/interfaces.go

```diff
@@ -82,6 +91,12 @@ func (client *Client) Interfaces() (interfaces Interfaces, err error) {
 	return
 }
 
+// InterfaceInfos returns all interfaces, their meta-data, plugs and slots.
+func (client *Client) InterfaceInfos() (infos map[string]InterfaceInfo, err error) {
```

Comment:
On second thought. How would this work with the current implementation that just displays a list of name + summary? We'd have to make 1+N calls to do that (one to get the names, then one for each name).

How about we return some basic meta-data (summary) alongside the name from InterfaceNames()? (perhaps it should be renamed to be appropriate)